### PR TITLE
[AIRFLOW-6307] Improve support for a special characters in DAG's name in stats view

### PR DESCRIPTION
**Pull request by mik-laj**
Special characters are not properly handled. It can cause problem.

---

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by ashb** [comment]
Is a comma actually allowed in a dag ID? (Do we want to allow it?)

File: airflow/www/views.py [hunk 1]
```diff
@@ -307,7 +307,7 @@ def dag_stats(self, session=None):
 
         # Filter by get parameters
         selected_dag_ids = {
-            unquote(dag_id) for dag_id in request.args.get('dag_ids', '').split(',') if dag_id
+            unquote(dag_id) for dag_id in request.args.getlist('dag_ids[]') if dag_id
```
Comment (ashb) [inline airflow/www/views.py:310]:
```suggestion
            unquote(dag_id) for dag_id in request.args.getlist('dag_ids') if dag_id
```

There's no requirement on `[]` got `getlist`, and both of those characters get url-escaped which just makes the URL a bit ugly is all (and longer) :)

**Comment by kaxil** [comment]
> Is a comma actually allowed in a dag ID? (Do we want to allow it?)

Wr definitely don't want to allow comma.

File: airflow/www/views.py (hunk 1 above)
Comment (potiuk) [reply airflow/www/views.py:310]: +1 -> [] is just a strange convention and not a widely used one. I tried to find anyone using it and could not find it..

Comment (mik-laj) [reply airflow/www/views.py:310]: PHP uses this syntax, but it's not very popular in the Python world. I will delete it.
https://stackoverflow.com/a/9547490/1849502

Comment (mik-laj) [reply airflow/www/views.py:310]: I found the real specifications that describe this format.
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#style-examples last row.
Screenshot 2020-02-03 at 21 39 32

**Pull request closed without merging**